libs/ledger-live-common/src/deposit/useGroupedCurrenciesByProvider.hook.ts

```diff
@@ -12,18 +12,30 @@ const initialResult: GroupedCurrencies = {
   currenciesByProvider: [],
 };
 
-export const useGroupedCurrenciesByProvider = () => {
+export const useGroupedCurrenciesByProvider = (
+  withLoading?: boolean,
+): GroupedCurrencies | LoadingBasedGroupedCurrencies => {
   const [result, setResult] = useState(initialResult);
 
+  const [loadingStatus, setLoadingStatus] = useState<LoadingStatus>(LoadingStatus.Idle);
   const coinsAndTokensSupported = useMemo(
     () => (listSupportedCurrencies() as CryptoOrTokenCurrency[]).concat(listSupportedTokens()),
     [],
   );
 
   // Get mapped assets filtered by supported & sorted currencies, grouped by provider id
   useEffect(() => {
-    loadCurrenciesByProvider(coinsAndTokensSupported).then(setResult);
-  }, [coinsAndTokensSupported]);
-
-  return result;
+    if (withLoading) {
+      setLoadingStatus(LoadingStatus.Idle);
+      loadCurrenciesByProvider(coinsAndTokensSupported)
+        .then(data => {
+          setResult(data);
+          setLoadingStatus(LoadingStatus.Success);
+        })
+        .catch(() => setLoadingStatus(LoadingStatus.Error));
+    } else {
+      loadCurrenciesByProvider(coinsAndTokensSupported).then(setResult);
+    }
+  }, [coinsAndTokensSupported, withLoading]);
+  return withLoading ? { result, loadingStatus } : result;
 };
```

Comment:
You can use TanStack Query to do that ?

Reply:
within the scope of this PR i juste used the existing promise, sure we can do it with tanstack query, react query mutations or any other async operation library. but it's not the scope of this ticket.